app/src/main/java/com/jerboa/db/AppDBMigrations.kt

```diff
@@ -469,6 +470,81 @@ val MIGRATION_30_31 =
+                      `use_custom_tabs` INTEGER NOT NULL DEFAULT ${DEFAULT_USE_CUSTOM_TABS},
+                      `use_private_tabs` INTEGER NOT NULL DEFAULT ${DEFAULT_USE_PRIVATE_TABS},
+                      `secure_window` INTEGER NOT NULL DEFAULT ${DEFAULT_SECURE_WINDOW},
+                      `blur_nsfw` INTEGER NOT NULL DEFAULT ${DEFAULT_BLUR_NSFW},
+                      `show_text_descriptions_in_navbar` INTEGER NOT NULL DEFAULT ${DEFAULT_SHOW_TEXT_DESCRIPTIONS_IN_NAVBAR},
+                      `markAsReadOnScroll` INTEGER NOT NULL DEFAULT ${DEFAULT_MARK_AS_READ_ON_SCROLL},
+                      `backConfirmationMode` INTEGER NOT NULL DEFAULT ${DEFAULT_BACK_CONFIRMATION_MODE},
+                      `show_post_link_previews` INTEGER NOT NULL DEFAULT ${DEFAULT_SHOW_POST_LINK_PREVIEWS},
+                      `post_actionbar_mode` INTEGER NOT NULL DEFAULT ${DEFAULT_POST_ACTION_BAR_MODE},
+                      `auto_play_gifs` INTEGER NOT NULL DEFAULT ${DEFAULT_AUTO_PLAY_GIFS},
+                      `swipe_to_action_preset` INTEGER NOT NULL DEFAULT ${DEFAULT_SWIPE_TO_ACTION_PRESET},
+                      `last_version_code_viewed` INTEGER NOT NULL DEFAULT $DEFAULT_LAST_VERSION_CODE_VIEWED
+                    )
+                """.trimIndent(),
+            )
+
+            // Need to select explicitly, because mark_changelog_viewed was dropped
+            db.execSQL(
+                """
+            INSERT INTO AppSettingsBackup SELECT
+                      `id`,
+                      `font_size`,
+                      `theme`,
+                      `theme_color`,
+                      `post_view_mode`,
+                      `show_bottom_nav`,
+                      `post_navigation_gesture_mode`,
+                      `show_collapsed_comment_content`,
+                      `show_comment_action_bar_by_default`,
+                      `show_voting_arrows_in_list_view`,
+                      `show_parent_comment_navigation_buttons`,
+                      `navigate_parent_comments_with_volume_buttons`,
+                      `use_custom_tabs`,
+                      `use_private_tabs`,
+                      `secure_window`,
+                      `blur_nsfw`,
+                      `show_text_descriptions_in_navbar`,
+                      `markAsReadOnScroll`,
+                      `backConfirmationMode`,
+                      `show_post_link_previews`,
+                      `post_actionbar_mode`,
+                      `auto_play_gifs`,
+                      `swipe_to_action_preset`,
+                      `last_version_code_viewed`
+             FROM AppSettings
+            """,
+            )
+            db.execSQL("DROP TABLE AppSettings")
+            db.execSQL("ALTER TABLE AppSettingsBackup RENAME to AppSettings")
+
+            // Reset a few messups to default
+            db.execSQL("UPDATE AppSettings SET post_actionbar_mode = $DEFAULT_POST_ACTION_BAR_MODE")
```

Comment:
There are probably more but only the enums matter, bc if they get assigned a value bigger than their enum size (due swapping between two enums which are bigger and that user having the biggest value) it would cause a NPE when we select the enum.

Reply:
Imo this is enough unless you want to reset more enums.